### PR TITLE
Handle DateTime objects in formatted messages

### PR DESCRIPTION
**Pull request by pierswarmers**
Adds support for DateTime objects being passed and formatted in PSRLogMessages.

Seems like it might be a good addition. Totally understand if this change pushes this processor beyond it's intended scope.

Cheers.

### CODE REVIEW

File: src/Monolog/Processor/PsrLogMessageProcessor.php [hunk 1]
```diff
@@ -33,7 +45,9 @@ public function __invoke(array $record): array
         $replacements = [];
         foreach ($record['context'] as $key => $val) {
             if (is_null($val) || is_scalar($val) || (is_object($val) && method_exists($val, "__toString"))) {
-                $replacements['{'.$key.'}'] = $val;
+                $replacements['{' . $key . '}'] = $val;
+            } elseif ($val instanceof \DateTime) {
```
Comment (stof) [inline src/Monolog/Processor/PsrLogMessageProcessor.php]: should check for DateTimeInterface too, to support DateTimeImmutable

File: src/Monolog/Processor/PsrLogMessageProcessor.php [hunk 2]
```diff
@@ -33,7 +45,9 @@ public function __invoke(array $record): array
         $replacements = [];
         foreach ($record['context'] as $key => $val) {
             if (is_null($val) || is_scalar($val) || (is_object($val) && method_exists($val, "__toString"))) {
-                $replacements['{'.$key.'}'] = $val;
+                $replacements['{' . $key . '}'] = $val;
```
Comment (stof) [inline src/Monolog/Processor/PsrLogMessageProcessor.php]: why are you changing coding standards ?

File: src/Monolog/Processor/PsrLogMessageProcessor.php [hunk 3]
```diff
@@ -20,6 +20,18 @@
  */
 class PsrLogMessageProcessor
 {
+    const SIMPLE_DATE = "Y-m-d\TH:i:sP";
+
+    protected $dateFormat;
```
Comment (stof) [inline src/Monolog/Processor/PsrLogMessageProcessor.php]: should be private IMO

**Comment by Seldaek** [comment]
Seems fine to me but yes as master target php7 this should be checking for DateTimeInterface. Please fix the other comments from @stof too, then it should be mergeable :)

**Comment by pierswarmers** [comment]
Thanks for the feedback! Changes have been made :)

**Comment by Seldaek** [comment]
:+1: Thanks

**Pull request merged**